build-system/tasks/amp-task-runner.js

```diff
@@ -165,7 +180,7 @@ function finalizeRunner() {
     log(red('ERROR:'), 'Unknown task', cyan(args.join(' ')));
     log('⤷ Run', cyan('amp --help'), 'for a full list of tasks.');
     log('⤷ Run', cyan('amp <task> --help'), 'for help with a specific task.');
-    process.exitCode = 1;
+    process.exit(1);
```

Comment:
Why this change?

Reply:
We started setting `exitCode` instead of directly exiting because `gulp` used to complain about unresolved promises and an uncalled `done()`. That's no longer a problem now.

Reply:
IMO it's better to set the exit code and letting the process exit gracefully. Because with `process.exit()`, there could be async callbacks or other stragglers that are terminated unexpectedly.

Reply:
After looking at the what else is done by `commander` in this scenario, I've reverted the change and gone back to setting `exitCode`.

However, there do remain other places where a hard exit is necessary. E.g. The task runner notices that the task implementation has encountered a fatal error, and waiting for graceful exit could result in the task stalling and never exiting. In those cases, the only way out is to call `process.exit(1)`.